runtime/common/metering.go

```diff
@@ -50,6 +50,27 @@ func NewConstantMemoryUsage(kind MemoryKind) MemoryUsage {
 	}
 }
 
+func NewArrayMemoryUsage(length int) MemoryUsage {
+	return MemoryUsage{
+		Kind:   MemoryKindArray,
+		Amount: uint64(length) + 1, // +1 to account for empty arrays
```

Comment:
I thought us doing this in the strings was messing up your calculations? Or is it fine?

My approach was to add another MemoryKind so every container would have two: a constant one for its existence and a variable one for its length.

Reply:
Agreed, it might be better to split this up. The code reads fine here, but is confusing below:
https://github.com/onflow/cadence/pull/1546/files#diff-424575c293dc2bc3a28aab845829d7656fa83a7be4e026de8906935e23bbd52eR1652

> common.NewArrayMemoryUsage(0)

This reads like the cost of allocating an "empty array", but really is the allocation cost of the addition of an element.